cdc/sinkv2/eventsink/event_appender.go

```diff
@@ -42,10 +46,60 @@ var _ Appender[*model.SingleTableTxn] = (*TxnEventAppender)(nil)
 type TxnEventAppender struct{}
 
 // Append appends the given rows to the given txn buffer.
+// The callers of this function should **make sure** that
+// the commitTs and startTs of rows is **strictly increasing**.
+// 1. Txns ordered by commitTs and startTs.
+// 2. Rows grouped by startTs and big txn batch,
+// cause the startTs is the unique identifier of the transaction.
 func (t *TxnEventAppender) Append(
 	buffer []*model.SingleTableTxn,
 	rows ...*model.RowChangedEvent,
 ) []*model.SingleTableTxn {
-	// TODO implement me
-	panic("implement me")
+	for _, row := range rows {
+		// This means no txn is in the buffer.
+		if len(buffer) == 0 {
+			txn := &model.SingleTableTxn{
+				StartTs:   row.StartTs,
+				CommitTs:  row.CommitTs,
+				Table:     row.Table,
+				ReplicaID: row.ReplicaID,
+			}
+			txn.Append(row)
+			buffer = append(buffer, txn)
+			continue
+		}
+
+		currentLastTxn := buffer[len(buffer)-1]
```

Comment:
The semantics of `currentLastTxn` seem redundant, perhaps `preTxn` or `lastTxn` are appropriate names.

Reply:
Renamed! Thanks!